components/entitlement/org.wso2.carbon.identity.entitlement/src/main/java/org/wso2/carbon/identity/entitlement/dao/JDBCSimplePAPStatusDataHandler.java

```diff
@@ -0,0 +1,389 @@
+import static org.wso2.carbon.identity.entitlement.dao.DAOConstants.STATUS_COUNT;
+
+import static java.time.ZoneOffset.UTC;
+
+public class JDBCSimplePAPStatusDataHandler implements PAPStatusDataHandler {
+
+    private static final Log AUDIT_LOG = CarbonConstants.AUDIT_LOG;
+    private static final String AUDIT_MESSAGE
+            = "Initiator : %s | Action : %s | Target : %s | Data : { %s } | Result : %s ";
+    private int maxRecords;
+
+    /**
+     * init entitlement status data handler module.
+     *
+     * @param properties properties.
+     */
+    @Override
+    public void init(Properties properties) {
+
+        maxRecords = EntitlementUtil.getMaxNoOfStatusRecords();
+    }
+
+    /**
+     * Handles the status data.
+     *
+     * @param about         whether the status is about a policy or publisher.
+     * @param key           key value of the status.
+     * @param statusHolders <code>StatusHolder</code>.
+     * @throws EntitlementException throws, if fails to handle.
+     */
+    @Override
+    public void handle(String about, String key, List<StatusHolder> statusHolders) throws EntitlementException {
+
+        // If the action is DELETE_POLICY, delete the policy or the subscriber status
+        for (StatusHolder holder : statusHolders) {
+            if (EntitlementConstants.StatusTypes.DELETE_POLICY.equals(holder.getType())) {
+                deletePersistedData(about, key);
+                return;
+            }
+        }
+        persistStatus(about, key, statusHolders);
+    }
+
+    /**
+     * Returns status data.
+     *
+     * @param about        indicates what is related with this admin status action.
+     * @param key          key value of the status.
+     * @param type         admin action type.
+     * @param searchString search string for <code>StatusHolder</code>.
+     * @return An array of <code>StatusHolder</code>.
+     * @throws EntitlementException if fails.
+     */
+    @Override
+    public StatusHolder[] getStatusData(String about, String key, String type, String searchString)
+            throws EntitlementException {
+
+        String statusAboutType = EntitlementConstants.Status.ABOUT_POLICY.equals(about)
+                ? EntitlementConstants.Status.ABOUT_POLICY
+                : EntitlementConstants.Status.ABOUT_SUBSCRIBER;
+
+        List<StatusHolder> holders = getStatus(key, statusAboutType);
+        return EntitlementUtil.filterStatus(holders, searchString, about, type);
+    }
+
+    /**
+     * Deletes all status records.
+     *
+     * @param about whether the status is about a policy or publisher.
+     * @param key   key value of the status.
+     * @throws EntitlementException if fails to delete.
+     */
+    public void deletePersistedData(String about, String key) throws EntitlementException {
+
+        int tenantId = CarbonContext.getThreadLocalCarbonContext().getTenantId();
+        String query = EntitlementConstants.Status.ABOUT_POLICY.equals(about) ?
```

Comment:
Also, this one too

Reply:
but this one is very borderline, since the value of this one is a sql query.. so, its ok to keep as well..